docs/development.md

```diff
@@ -1,26 +1,254 @@
 # Development
 
-This section outlines how to approach the operations and development of the HTC Grid.
+This section outlines how to develop and deploy a custom application on HTC-Grid.
 
-![Test](./images/development.png)
+At the top level there are 3 main components that need to be developed:
 
-HTC-Grid comes with a mock execution engine mock_compute_engine.cpp. This is the executable that is invoked for every submitted task. In real deployments this executable will need to be replaced with a real implementation. Steps below outline how to modify and apply changes to the custom execution code.
+1. A **client application(s)** that will interact with a deployment of HTC-Grid by submitting tasks and retrieving results.
+2. A **worker lambda function** that will be receiving and executing tasks.
+3. **Configuration of the HTC-Grid's deployment process** to incorporate all the relevant changes (specifically for the backend worker lambda functions).
 
-## Building Lambda layer
 
-In order to apply changes of the lambda layer, you need to build lambda layer container and execute it locally.
+## 1. Developing a Client Application
 
-* Go to src_client folder. This folder contains all the files necessary for building lambda layer.
-* Build the docker image, note the ID of the created container.
+HTC-Grid implements Python3 API (please refer to [API Reference](docs/reference.md) for more details) and provides several client application examples of how to use the API (see `client.py`, `cancel_tasks.py`, and `portfolio_pricing_client.py`  examples in `/examples/client/python`).
+
+These examples show how to:
+- Connect to an HTC-Grid deployment
+- Authenticate the client
+- Submit multiple tasks
+- Wait for the results
+- Retrieve results
+
+Follow these examples to write a custom applications.
```

Comment:
Perhaps it would be useful to add here a code block with the main things that the client does ?